### PR TITLE
Update 9563

### PR DESCRIPTION
**Pull request by zhouying7780**
https://mastern-jenkins-csb-openshift-qe.cloud.paas.psi.redhat.com/job/Runner-v3/117745/console

@kasturinarra Please help review , thanks.

### CODE REVIEW

File: features/cli/deploy.feature [hunk 1]
```diff
@@ -347,6 +347,7 @@ Feature: deployment related features
       | replicas | 0                |
     Then the step should succeed
     Given I wait until number of replicas match "0" for replicationController "ab-example-a-1"
+    And I wait until the status of deployment "ab-example-b" becomes :complete
```
Comment (kasturinarra) [inline features/cli/deploy.feature:350]: @zhouying7780 i have seen in multiple cases that this line fails, may be better to add  as below ?

And I wait for the steps to pass
"""
oc get deployment ab-example-b and check ffor status 
"""

File: features/cli/deploy.feature [hunk 2]
```diff
@@ -361,6 +362,7 @@ Feature: deployment related features
       | replicas | 1                |
     Then the step should succeed
     Given I wait until number of replicas match "0" for replicationController "ab-example-b-1"
+    Given I wait until number of replicas match "1" for replicationController "ab-example-a-1"
```
Comment (kasturinarra) [inline features/cli/deploy.feature:365]: Also this line, may be it is better to add as below ?

Given number of replicas of "ab-example-a-1" deployment becomes:
      | current | 1 |

Comment (kasturinarra) [reply features/cli/deploy.feature:365]: @zhouying7780  ^^

**Comment by zhouying7780** [comment]
@kasturinarra  Have resolved all the comments, please help review again, thanks.

**Comment by kasturinarra** [comment]
@zhouying7780 could you please help me understand what has been changed here ? Thanks !!

**Comment by zhouying7780** [comment]
@kasturinarra  Please help review again, have resolved all the comments. Thanks.

File: features/cli/deploy.feature [hunk 3]
```diff
@@ -363,7 +363,16 @@ Feature: deployment related features
...
+    Given number of replicas of "ab-example-a" deployment config becomes:
+      | desired   | 0 |
+      | current   | 0 |
+      | updated   | 0 |
+      | available | 0 |
+    Given number of replicas of "ab-example-b" deployment config becomes:
+      | desired   | 1 |
+      | current   | 1 |
+      | updated   | 1 |
+      | available | 1 |
```
Comment (kasturinarra) [inline features/cli/deploy.feature:375]: @zhouying7780 do we need steps from 371 to 375 ?  I understand that we here are making sure to have replicas as '0' right.

File: features/cli/deploy.feature [hunk 4]
```diff
@@ -377,8 +386,16 @@ Feature: deployment related features
...
+    Given number of replicas of "ab-example-b" deployment config becomes:
+      | desired   | 0 |
+      | current   | 0 |
+      | updated   | 0 |
+      | available | 0 |
+    Given number of replicas of "ab-example-a" deployment config becomes:
+      | desired   | 1 |
+      | current   | 1 |
+      | updated   | 1 |
+      | available | 1 |
```
Comment (kasturinarra) [inline features/cli/deploy.feature:398]: Do we need line no: 389 to 393 ? I understand that we are waiting for replicas to become 1.

**Comment by kasturinarra** [comment]
@zhouying7780 please help resolve the comments here, thanks !!

**Comment by zhouying7780** [comment]
@kasturinarra sorry for so late , have resolved all the comments, please help review again, thanks.

File: features/cli/deploy.feature [hunk 5]
```diff
@@ -377,8 +381,11 @@ Feature: deployment related features
       | name     | ab-example-a     |
       | replicas | 1                |
     Then the step should succeed
```
Comment (kasturinarra) [inline features/cli/deploy.feature:383]: @zhouying7780 Do we need this line ? I think we can remove it, what do you say ?

Comment (zhouying7780) [reply features/cli/deploy.feature:383]: @kasturinarra   the scale active is needed.

Comment (kasturinarra) [reply features/cli/deploy.feature:383]: @zhouying7780 okay, thanks !! Got it

**Comment by kasturinarra** [comment]
@liangxia could you please help review & merge ? Thanks !!

**Pull request merged**